doc/api/timers.md

```diff
@@ -2,87 +2,139 @@
 
-Stops an `immediateObject`, as created by [`setImmediate`][], from triggering.
+## Class: Immediate
 
-## clearInterval(intervalObject)
+This object is created internally and is returned from [`setImmediate`][]. It
+can be passed to [`clearImmediate`] in order to cancel the scheduled actions.
 
-Stops an `intervalObject`, as created by [`setInterval`][], from triggering.
+## Class: Timeout
 
-## clearTimeout(timeoutObject)
+This object is created internally and is returned from [`setTimeout`][] and
+[`setInterval`][]. It can be passed to [`clearTimeout`][] or [`clearInterval`][]
+respectively in order to cancel the scheduled actions.
 
-Prevents a `timeoutObject`, as created by [`setTimeout`][], from triggering.
+By default, when a timer is scheduled using either [`setTimeout`] or
+[`setInterval`][], the Node.js event loop will continue running as long as the
+timer is active. Each of the `Timeout` objects returned by these functions
+export both `timer.ref()` and `timer.unref()` functions that can be used to
+control this default behavior.
 
-## ref()
+### timeout.ref()
 
-If a timer was previously `unref()`d, then `ref()` can be called to explicitly
-request the timer hold the program open. If the timer is already `ref`d calling
-`ref` again will have no effect.
+When called, requests that the Node.js event loop *not* exit so long as the
+timer is active. Calling `timer.ref()` multiple times will have no effect.
 
-Returns the timer.
+Returns a reference to the `Timeout`.
 
-## setImmediate(callback[, arg][, ...])
+### timeout.unref()
 
-Schedules "immediate" execution of `callback` after I/O events'
-callbacks and before timers set by [`setTimeout`][] and [`setInterval`][] are
-triggered. Returns an `immediateObject` for possible use with
-[`clearImmediate`][]. Additional optional arguments may be passed to the
-callback.
+When called, allows the Node.js event loop to exit if the timer is the only
+item left in the Node.js event loop. Calling `timer.unref()` multiple times
+will have no effect.
 
-Callbacks for immediates are queued in the order in which they were created.
-The entire callback queue is processed every event loop iteration. If an
-immediate is queued from inside an executing callback, that immediate won't fire
-until the next event loop iteration.
+In the case of [`setTimeout`][], `timer.unref()` creates a separate timer that
+will wake the Node.js event loop. Creating too many of these can adversely
+impact the performance.
 
-If `callback` is not a function `setImmediate()` will throw immediately.
+Returns a reference to the `Timeout`.
 
-## setInterval(callback, delay[, arg][, ...])
+## Scheduling Timers
+
+A timer in Node.js is an internal construct that calls a given function after
+a certain period of time. When a timer's function is called is a factor of
+which method was used to create the timer and what other work the Node.js
+event loop is doing.
+
+### setImmediate(callback[, arg][, ...])
```

Comment:
I still think this should be `[, ...args]` (i.e. spread/rest params notation) but since that will change the links maybe we should discuss it in a separate issue.